crates/lib3h/src/engine/real_engine.rs

```diff
@@ -554,6 +503,58 @@ impl<T: Transport, D: Dht, SecBuf: Buffer, Crypto: CryptoSystem> RealEngine<T, D
         Ok(output)
     }
 
+    fn serve_DirectMessage(
+        &mut self,
+        msg: DirectMessageData,
+        is_response: bool,
+    ) -> Lib3hServerProtocol {
+        // Check if space is joined by sender
+        let maybe_space = self.get_space_or_fail(
+            &msg.space_address,
+            &msg.from_agent_id,
+            &msg.request_id,
+            None,
+        );
+        // Return failure if not
+        if let Err(failure_msg) = maybe_space {
+            return failure_msg;
+        }
+        let space_gateway = maybe_space.unwrap();
+        // Prepare response
+        let mut response = GenericResultData {
+            request_id: msg.request_id.clone(),
+            space_address: msg.space_address.clone(),
+            to_agent_id: msg.from_agent_id.clone(),
+            result_info: vec![],
+        };
+        // Check if messaging self
+        let this_peer = space_gateway.this_peer();
+        if this_peer.peer_address.as_bytes() == msg.to_agent_id.as_slice() {
+            response.result_info = "Messaging self".as_bytes().to_vec();
+            return Lib3hServerProtocol::FailureResult(response);
```

Comment:
Why is messaging yourself considered a failure?

Reply:
Because the message will not be sent since you dont have a connection id to self, and there is no use-case for this. Allowing it would be a waste of cycles.